### PR TITLE
Bump version to 1.0.0 globally

### PR DESCRIPTION
**Pull request by gavofyork**
Also some minor changes to staking/phragmen:

- Don't pass a `Fn()->usize`, only to always call it once: just pass a `usize`!
- Rename `rounds` (which is incredibly opaque jargon taking from Phragmen's original context) to `validator_count`.
- `min_validator_count` represents the minimum number of validators we need, not one-less, so we require the number of validators to be `>=` before we declare code-red, not `>`.
- `validator_count` represents the ideal validator count. We only need to do candidate selection if we have more than that, so the comparison there should be `>`, not `>=`.
- Update tests accordingly.

### CODE REVIEW

File: srml/staking/src/phragmen.rs [hunk 1]
```diff
@@ -109,9 +108,8 @@ pub fn elect<T: Trait + 'static, FR, FN, FV, FS>(
 	>>,
 	for <'r> FS: Fn(&'r T::AccountId) -> BalanceOf<T>,
 {
-	let expand = |b: BalanceOf<T>|    <T::CurrencyToVote as Convert<BalanceOf<T>, u64>>::convert(b) as ExtendedBalance;
```
Comment (gavofyork) [inline srml/staking/src/phragmen.rs]: CC @Kianenigma : No extraneous whitespace within code.

File: srml/staking/src/phragmen.rs [hunk 2]
```diff
@@ -92,15 +92,14 @@ pub struct Edge<AccountId> {
...
 /// Returns None if not enough candidates exist.
-pub fn elect<T: Trait + 'static, FR, FN, FV, FS>(
-		get_rounds: FR,
-		get_validators: FV,
-		get_nominators: FN,
-		stash_of: FS,
-		minimum_validator_count: usize,
-		config: ElectionConfig<BalanceOf<T>>,
+pub fn elect<T: Trait + 'static, FN, FV, FS>(
+	validator_count: usize,
```
Comment (gavofyork) [inline srml/staking/src/phragmen.rs]: rename `rounds` to `validator_count` - to minimise use of jargon.

also wtf is this passing in a closure only to call it immediately?

Comment (Kianenigma) [reply srml/staking/src/phragmen.rs]: See the open discussions here. https://github.com/paritytech/substrate/pull/1915#discussion_r263425454

I would personally not do that and pre-process the data in `lib.re` and pass everything in as a vectors.

**Comment by gavofyork** [comment]
done.

**Pull request merged**